server/src/main/java/org/apache/iotdb/db/metadata/template/Template.java

```diff
@@ -30,49 +39,70 @@
 
 import org.apache.commons.lang3.builder.HashCodeBuilder;
 
+import java.util.ArrayDeque;
 import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Deque;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
+
 public class Template {
   private String name;
 
-  private Map<String, IMeasurementSchema> schemaMap = new HashMap<>();
+  private IMNode mountedNode;
+  private Set<String> alignedPrefix;
+  private List<String> mountedPath;
```

Comment:
It is designed that one template can be set on one mnode and the descendants of this mnode can use this template, which we call template instantiation. 
I think the mountedNode in your code is the mnode this template set to and the mountedPath represents the mnode using this template. 
It's better to use different terms to distinguish such two kinds mnodes. Or generating your mountedPath at runtime via a new traverse method in MTree.

Reply:
renamed to templateRoot now